deploy/manifests/scc.yaml

```diff
@@ -0,0 +1,37 @@
+# Create SCC for Gateway resources
+kind: SecurityContextConstraints
+apiVersion: security.openshift.io/v1
+metadata:
+  name: nginx-ngf-admin
+allowPrivilegedContainer: false
```

Comment:
Do we need to specify values when they are false?

Reply:
Nope! should just be a boolean shown here: https://docs.okd.io/4.15/rest_api/security_apis/securitycontextconstraints-security-openshift-io-v1.html

Reply:
Sorry, I mean can we omit these fields since their values are false? Isn't the default value false?

Reply:
Hmmm, so as seen in the docs here: https://docs.openshift.com/container-platform/4.15/rest_api/security_apis/securitycontextconstraints-security-openshift-io-v1.html

it has a list of **required** fields, which is why I decided to include them. But I just tested it and I was able to omit those fields with NGF still deploying correctly.

What would you recommend?

Reply:
> A security context (SC), defined in a pod, enables a deployer to specify a container's permissions to access protected functions. When the pod creates the container, it configures the container to allow these permissions and block all others. The cluster will only deploy the pod if the permissions it requests are allowed by a corresponding SCC.

[source](https://developer.ibm.com/learningpaths/secure-context-constraints-openshift/intro/)

IMO the SCC should only list the permissions that the pod needs, not the permissions a pod doesn't need. 

@sjberman @ciarams87 thoughts?

Reply:
removed the fields, will leave conversation open for @ciarams87 to provide thoughts if she wants

Reply:
Update:

When I git clone NGF into the Openshift environment. Add the `scc.yaml` template. Make the changes to the `values.yaml` and `rbac.yaml`, then try to deploy NGF using Helm with the edited real chart by running 

`helm install ngf ./charts/nginx-gateway-fabric --create-namespace -n nginx-gateway --set nginxGateway.productTelemetry.enable=false --set platform=openshift`

I get this error

`
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: error validating "": error validating data: [ValidationError(SecurityContextConstraints): missing required field "allowHostDirVolumePlugin" in io.openshift.security.v1.SecurityContextConstraints, ValidationError(SecurityContextConstraints): missing required field "allowHostIPC" in io.openshift.security.v1.SecurityContextConstraints, ValidationError(SecurityContextConstraints): missing required field "allowHostNetwork" in io.openshift.security.v1.SecurityContextConstraints, ValidationError(SecurityContextConstraints): missing required field "allowHostPID" in io.openshift.security.v1.SecurityContextConstraints, ValidationError(SecurityContextConstraints): missing required field "allowHostPorts" in io.openshift.security.v1.SecurityContextConstraints, ValidationError(SecurityContextConstraints): missing required field "allowPrivilegedContainer" in io.openshift.security.v1.SecurityContextConstraints]
`

However, when I deploy NGF using the 1.2 chart like this 

`helm install ngf oci://ghcr.io/nginxinc/charts/nginx-gateway-fabric --create-namespace -n nginx-gateway --set nginxGateway.productTelemetry.enable=false`

Then manually make the changes to the clusterrole, and apply an `scc.yaml` file. It does not error. 

Even through restarts of the NGF deployment, it never errors. 

I have no idea why its enforcing only when I template the scc.yaml file and have it auto-deploy and not manually. Is it worth digging deeper in this rabbit hole or should I just add the fields back?

Reply:
@bjee19 
could you run `helm template` instead `helm install` with the same arguments to take a look at the generated manifests?

Reply:
Ran it and the scc.yaml looks correct:

```
---
# Source: nginx-gateway-fabric/templates/scc.yaml
# Create SCC for Gateway resources
kind: SecurityContextConstraints
apiVersion: security.openshift.io/v1
metadata:
  name: nginx-ngf-admin
allowPrivilegeEscalation: false
readOnlyRootFilesystem: true
runAsUser:
  type: MustRunAsRange
  uidRangeMin: 101
  uidRangeMax: 102
fsGroup:
  type: MustRunAs
  ranges:
    - min: 1001
      max: 1001
supplementalGroups:
  type: MustRunAs
  ranges:
    - min: 1001
      max: 1001
seLinuxContext:
  type: MustRunAs
volumes:
  - emptyDir
  - secret
users:
  - system:serviceaccount:*:ngf-nginx-gateway-fabric
allowedCapabilities:
  - NET_BIND_SERVICE
  - KILL
requiredDropCapabilities:
  - ALL


```

So does the rbac.yaml

Reply:
what if you try to apply the whole generated yaml? what if apply only SCC?

Reply:
Ah I think I found the issue, when running `oc create -f scc.yaml` with the above file, no error.

But when I run `kubectl apply -f scc.yaml` I get the error. All the Openshift documentation use `oc` when creating the `SCC` object, but I guess they have no validation on it, only the `kubectl` command.

Reply:
> To satisfy both, should I include the required fields?

probably. I'd suggest also searching how  any other projects do that if they do that

Reply:
@kate-osborn @sjberman  any thoughts?

I did a quick search on Github and found these other prominent examples which list the fields as false:

https://github.com/OpenShiftDemos/openshift-ops-workshops/blob/5d540ca583b43375cd80c76beee2e6a64edf0e8f/support/restrictedfsgroupscc.yaml#L4

https://github.com/grafana/loki/blob/88c671162f70e075f6aa43599aa560fe7b4b5627/production/helm/loki/templates/securitycontextconstraints.yaml#L9

https://github.com/Qovery/engine/blob/4cc39e0c827b3f98736f9646e397abf52e03defb/lib/common/bootstrap/charts/datadog/templates/agent-scc.yaml#L38

Understanding that being a popular project doesn't mean all the code is of best style and practices, it still seems that these fields are **required** and should be listed as "false" even when thats the default. Thoughts?

Reply:
Bummer. Thanks for trying. Including them is fine 👍